### PR TITLE
Implement Spec API to Docs Setup

### PR DESCRIPTION
**Pull request by webspaceadam**
## Summary

- [Ticket](#)
- [Design](#)
- [Storybook](#)

### General

- [ ] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [ ] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--page).
  - [ ] The code is formatted with Prettier.
  - [ ] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [ ] The changes do not contain any breaking changes.
- [ ] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [ ] The component documentation is updated.
- [ ] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--page).
  - [ ] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [ ] All tests are 🟢.
- [ ] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--page).
  - [ ] WCAG 2.1 Level A and Level AA requirements are met.
  - [ ] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--page).
  - [ ] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--page).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--page)

### CODE REVIEW

File: apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+  return {
+    fallback: false,
+    paths: specs,
+  };
+};
+
+function generateGeneralDescription(document: ApiDoc) {
+  return document.definition?.info.description
+    ?.replace(document.title, "")
+    .replace(document.shortDescription, "")
```
Comment (webspaceadam) [inline apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx:50]: headsup: i know this only works for now, but i don't want to make the PR bigger as it is to add a module to handle replacing and handling the strings in the markdown string for now. i hope this is ok 😅

File: apps/swirl-docs/src/lib/navigation/src/data/apiDocs.data.ts [hunk 2]
```diff
@@ -2,14 +2,207 @@ import { NavItem } from "../navigation.model";
```
Comment (Sqrrl) [inline apps/swirl-docs/src/lib/navigation/src/data/apiDocs.data.ts:2]: **question:** Can this be added to the gitignore? Seems to be generated and contains paths to your local file system that wouldn't work on other machines anyway.

File: apps/swirl-docs/src/components/CodePreview/CodePreview.tsx [hunk 3]
```diff
@@ -34,46 +40,57 @@ export function CodePreview({ children, codeExample }: CodePreviewProps) {
...
               >
-                {isCopied ? "Code copied!" : "Copy Code"}
-                {isCopied ? (
-                  <SwirlIconCheckStrong size={16} className="ml-1" />
-                ) : (
-                  <SwirlIconCopy size={16} className="ml-1" />
-                )}
-              </button>
-            </CopyToClipboard>
+                <button
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/CodePreview/CodePreview.tsx:76]: **issue:** Misses a label.

File: apps/swirl-docs/src/components/Layout/CategoryNav.tsx [hunk 4]
```diff
@@ -25,18 +31,84 @@ export function CategoryNav() {
...
+    const [isExpanded, setIsExpanded] = useState(false);
+
+    useEffect(() => {
+      if (activePath.includes(navItem.url)) {
+        setIsExpanded(true);
+      }
+    }, [navItem.url]);
+
+    return (
+      <li className={classNames("mb-2 py-2", { "max-h-10": !isExpanded })}>
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Layout/CategoryNav.tsx:44]: **issue:** See https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/ for collapsible navigation pattern. (aria-expanded, aria-controls).

File: apps/swirl-docs/src/components/Layout/CategoryNav.tsx [hunk 5]
```diff
@@ -25,18 +31,84 @@ export function CategoryNav() {
...
+        <div className="flex justify-between items-center">
+          <Link href={`${navItem.url}`}>
+            <a
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Layout/CategoryNav.tsx:47]: **nitpick:** Could you make the link fill the available width? There is an awkward gap between the link and the toggle which is non interactive.

File: apps/swirl-docs/src/components/Layout/CategoryNav.tsx [hunk 6]
```diff
@@ -25,18 +31,84 @@ export function CategoryNav() {
...
+                  "text-border-info": activePath === navItem.url,
+                  "text-text-default": activePath !== navItem.url,
+                }
+              )}
+            >
+              <span>{navItem.title}</span>
+            </a>
+          </Link>
+          {navItem.children && (
+            <button
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Layout/CategoryNav.tsx:61]: **issue:** Missing a label + see above regarding Disclosure pattern.

File: apps/swirl-docs/src/components/Layout/CategoryNav.tsx [hunk 7]
```diff
@@ -25,18 +31,84 @@ export function CategoryNav() {
...
+        {navItem.children && isExpanded && (
+          <ul className="mt-4">
+            {navItem.children?.map((child, index) => {
+              return (
+                <li
+                  key={index}
+                  className="flex items-center max-h-40 h-10 mb-4 ml-6"
+                >
+                  <Link href={`${child.url}`}>
+                    <a
```
Comment (Sqrrl) [inline apps/swirl-docs/src/components/Layout/CategoryNav.tsx:84]: **nitpick:** `aria-current` on the active link would be nice.

File: apps/swirl-docs/src/lib/docs/src/oasBuilder.ts [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+    };
+  }
+
+  public generateResponse(operation: Operation) {
+    // currently we just take the first element as our OA specs are not fully functional. E.g. 201 is not defined for post for some requests.
+    const responseExample = operation.getResponseExamples()[0].mediaTypes[
+      "application/json"
+    ] as Array<MediaTypeObject>;
+
+    // console.log(`RESPONSE EXAMPLE: `, operation.getResponseExamples());
```
Comment (Sqrrl) [inline apps/swirl-docs/src/lib/docs/src/oasBuilder.ts:220]: **nitpick:** remove

File: apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+        }}
+        disableToc
+        header={<DocumentationLayout.Header className="col-span-2" />}
+        content={
+          <>
+            <DocumentationLayout.MDX />
+            <div className="mt-20">
+              {document.endpoints?.map((endpoint, index) => {
+                return (
+                  <>
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx:196]: **issue:** Missing a key. You could use `<article aria-labelledby={endpoint.path.split("#")[1]}>` here, in addition.

File: apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+                    <h2
+                      className="text-font-size-2xl font-font-weight-semibold mb-4"
+                      id={endpoint.path.split("#")[1]}
+                      key={`endpoint.title${index}`}
```
Comment (Sqrrl) [inline apps/swirl-docs/src/pages/api-docs/[apiSpec].tsx:200]: **issue:** Unnecessary key here.

**Review by Sqrrl — CHANGES_REQUESTED** [review]
- Could we make the left navigation sticky?
- Code preview is broken with longer endpoints. We should make a layout fine tuning separately.

Bildschirmfoto 2023-05-03 um 10 43 37

File: apps/swirl-docs/src/lib/navigation/src/data/apiDocs.data.ts (hunk 2 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/lib/navigation/src/data/apiDocs.data.ts:2]: good catch!

File: apps/swirl-docs/src/lib/docs/src/oasBuilder.ts (hunk 8 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/lib/docs/src/oasBuilder.ts:220]: done

File: apps/swirl-docs/src/components/Layout/CategoryNav.tsx (hunk 5 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/components/Layout/CategoryNav.tsx:47]: Bildschirm­foto 2023-05-03 um 16 15 26

do you mean this one?

**Comment by webspaceadam** [comment]
Ticket for Layout [Finetuning](https://flipapp.myjetbrains.com/issue/APP-14947/Layout-fine-tuning)

File: apps/swirl-docs/src/components/Layout/CategoryNav.tsx (hunk 5 above)
Comment (webspaceadam) [reply apps/swirl-docs/src/components/Layout/CategoryNav.tsx:50]: ah that link! alright :)

**Review by Sqrrl — APPROVED** [review]
